network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/detail_page.py

```diff
@@ -115,64 +107,144 @@ class RCCDetailPage(BasePage):
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
+        index.RelatedFields(
+            "rcc_authors",
+            [
+                index.RelatedFields(
+                    "author_profile",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_content_types",
+            [
+                index.RelatedFields(
+                    "content_type",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_curricular_areas",
+            [
+                index.RelatedFields(
+                    "curricular_area",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_topics",
+            [
+                index.RelatedFields(
+                    "rcc_topic",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
     ]
 
-    # def get_context(self, request):
-    #     context = super().get_context(request)
-    #     context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
-    #     context["rcc_authors"] = self.get_rcc_authors()
-    #     context["rcc_author_names"] = self.get_rcc_author_names()
-    #     context["content_type_names"] = self.get_related_content_types_names()
-    #     return context
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
+        context["rcc_authors"] = self.get_rcc_authors()
+        context["rcc_author_names"] = self.get_rcc_author_names()
+        context["content_type_names"] = self.get_related_content_types_names()
+        return context
 
-    # def get_rcc_authors(self):
-    #     rcc_author_profiles = localize_queryset(
-    #         Profile.objects.prefetch_related("authored_rcc_entries").filter(authored_rcc_entries__rcc_detail_page=self)
-    #     )
-    #     return rcc_author_profiles
+    def get_rcc_authors(self):
+        rcc_author_profiles = localize_queryset(
+            Profile.objects.prefetch_related("authored_rcc_articles").filter(
+                authored_rcc_articles__rcc_detail_page=self
+            )
+        )
+        return rcc_author_profiles
 
-    # def get_rcc_author_names(self):
-    #     return [ra.author_profile.name for ra in self.rcc_authors.all()]
+    def get_rcc_author_names(self):
+        return [ra.author_profile.name for ra in self.rcc_authors.all()]
 
-    # def get_related_content_types_names(self):
-    #     return [ct.content_type.name for ct in self.related_content_types.all()]
+    def get_related_content_types_names(self):
+        return [ct.content_type.name for ct in self.related_content_types.all()]
```

Comment:
**Question** This seems to pull the strings for display. Content types seem like they might be translatable. This is I think one of the use cases where it would be good to get the queryset of the actual objects (the content types) then localize those and then use the attribute (here the name) from the localized objects. 

I believe there was a utility that was created for this.

Reply:
#10017 This was not marked as RCC. I'll put it as clean up after the MVP.